packages/translated-components/src/index.js

```diff
@@ -61,16 +62,29 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
+const templateIteratee = (values, acc, v, k) => {
+  if (typeof v === 'object' && !v.format) {
+    acc[k] = reduce(v, partial(templateIteratee, values), {})
+  } else {
+    acc[k] = v.format(values)
```

Comment:
Wouldn't this break whenever a value is _not_ an object with a format method? 

I'm trying to get a little context from: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Text_formatting

This looks a little like a gotcha, a sub-object could have a key `format` which is itself to be translated, not a `NumberFormat` instance etc. You'd get a `v.format is a not a function` error, might take some digging.